### PR TITLE
rollback CoC

### PR DESCRIPTION
**Pull request by ivy-rew**

### CODE REVIEW

File: github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  private void removeFile(GHRepository repo, GHContent foundFile) throws IOException {
+    var defaultBranch = repo.getBranch(repo.getDefaultBranch());
+    var sha1 = defaultBranch.getSHA1();
+    repo.createRef("refs/heads/" + reference.meta().branchName(), sha1);
+    foundFile.delete(reference.meta().commitMessage(), reference.meta().branchName());
+    var pr = repo.createPullRequest(reference.meta().pullRequestTitle(), reference.meta().branchName(), repo.getDefaultBranch(), "");
+    LOG.info("Review the PR on "+ pr.getUrl());
+    //pr.merge(reference.meta().commitMessage());
```
Comment (ivy-rew) [inline github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java:106]: not merging ... yet

File: github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  }
+
+  public int removeFile(List<String> orgNames) throws IOException {
+    Objects.requireNonNull(orgNames);
+    var github = GitHubProvider.get();
+    LOG.info("Working on organizations: {0}.", orgNames);
+    for (var orgName : orgNames) {
+      var org = github.getOrganization(orgName);
+      for (var repo : List.copyOf(org.getRepositories().values())) {
+        missingFile(repo);
```
Comment (weissreto) [inline github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java:36]: removeFile

File: github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      LOG.info("Repo {0} is {1}.", repo.getFullName(), repo.isPrivate() ? "private" : "archived");
+      return;
+    }
+
+    var foundFile = getFileContent(reference.meta().filePath(), repo);
+    if (foundFile != null) {
+      if (hasSimilarContent(foundFile)) {
+        LOG.info("Repo {0} contains {1}", repo.getFullName(), foundFile.getName());
+        isNotSync = true;
+        handleFile(repo, foundFile);
```
Comment (weissreto) [inline github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java:62]: removeFile

File: github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  private void handleFile(GHRepository repo, GHContent foundFile) throws IOException {
+    try {
+      if (DryRun.is()) {
+        isNotSync = true;
+        LOG.info("DRYRUN: ");
+      } else {
+        removeFile(repo, foundFile);
+      }
+      LOG.info("Repo {0} {1} synced.", repo.getFullName(), reference.meta().filePath());
```
Comment (weissreto) [inline github-repo-manager/src/main/java/com/axonivy/github/file/GitHubFilesRemover.java:92]: Repo {0} file {1} removed

**Review by weissreto — APPROVED** [review]
Some comments

File: github-repo-manager/src/main/java/com/axonivy/github/file/GitHubMissingFilesDetector.java [hunk 5]
```diff
@@ -102,9 +107,16 @@ private void addMissingFile(GHRepository repo) throws IOException {
...
-        .message(reference.meta().commitMessage()).commit();
+    repo.createContent()
+      .branch(reference.meta().branchName())
+      .path(reference.meta().filePath())
+      .content(reference.content())
+      .message(reference.meta().commitMessage())
+      .commit();
     var pr = repo.createPullRequest(reference.meta().pullRequestTitle(), reference.meta().branchName(), repo.getDefaultBranch(), "");
+    if (ghActor != null) {
+      pr.setAssignees(ghActor);
```
Comment (ivy-rew) [inline github-repo-manager/src/main/java/com/axonivy/github/file/GitHubMissingFilesDetector.java:118]: even if we do an 'auto-merge' the pipeline now documents the running as assignee ... this allows users to negotiate this PR even back in time.

**Pull request merged**